sql/core/src/test/scala/org/apache/spark/sql/execution/SparkSqlParserSuite.scala

```diff
@@ -330,4 +330,42 @@ class SparkSqlParserSuite extends AnalysisTest {
     assertEqual("ADD FILE /path with space/abc.txt", AddFileCommand("/path with space/abc.txt"))
     assertEqual("ADD JAR /path with space/abc.jar", AddJarCommand("/path with space/abc.jar"))
   }
+
+  test("SPARK-32607: Script Transformation ROW FORMAT DELIMITED" +
+    " `TOK_TABLEROWFORMATLINES` only support '\\n'") {
+
+      // test input format TOK_TABLEROWFORMATLINES
+      intercept(
+          s"""
+             |SELECT TRANSFORM(a, b, c, d, e)
+             |  ROW FORMAT DELIMITED
+             |  FIELDS TERMINATED BY ','
+             |  LINES TERMINATED BY '@'
+             |  NULL DEFINED AS 'null'
+             |  USING 'cat' AS (value)
+             |  ROW FORMAT DELIMITED
+             |  FIELDS TERMINATED BY '&'
+             |  LINES TERMINATED BY '\n'
```

Comment:
If it only supports one value, why do we ever provide a clause to set it?

Reply:
> If it only supports one value, why do we ever provide a clause to set it?

Perhaps it's because previous developers have followed hive's approach.

Reply:
what's the drawback if we don't forbid arbitrary line terminator?

Reply:
> what's the drawback if we don't forbid arbitrary line terminator?

If user use wrong delimit , will got wrong result since in no-serde mode, we read output line by '\n'